src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostBuilder.cs

```diff
@@ -168,6 +168,11 @@ public void ConfigureContainer<TBuilder>(IServiceProviderFactory<TBuilder> facto
             };
         }
 
+        public void SetLazyLoadDefinition(WebAssemblyLazyLoadDefinition lazyLoadDefinition)
+        {
+            WebAssemblyNavigationManager.Instance.LazyLoadDefinition = lazyLoadDefinition;
+        }
```

Comment:
I think the layering is not right here. 

The mapping between routes and assemblies should be a responsibility of the router, not the host (core) framework.
The host should only be responsible for offering functionality to try and load a given resource lazily by name.